src/h5web/visualizations/heatmap/HeatmapToolbar.tsx

```diff
@@ -1,46 +1,50 @@
-import React from 'react';
-import shallow from 'zustand/shallow';
+import React, { ReactElement } from 'react';
+import { MdAspectRatio, MdGridOn, MdLinearScale } from 'react-icons/md';
 import ColorMapSelector from './ColorMapSelector';
 import Toggler from '../shared/Toggler';
 import { useHeatmapConfig } from './config';
 import DomainSlider from './DomainSlider';
 import ScreenshotButton from '../shared/ScreenshotButton';
+import Separator from '../shared/Separator';
+import Toolbar from '../shared/Toolbar';
 
-function HeatmapToolbar(): JSX.Element {
-  const [
+function HeatmapToolbar(): ReactElement {
+  const {
     hasLogScale,
     toggleLogScale,
     keepAspectRatio,
     toggleAspectRatio,
     showGrid,
     toggleGrid,
-  ] = useHeatmapConfig(state => [
-    state.hasLogScale,
-    state.toggleLogScale,
-    state.keepAspectRatio,
-    state.toggleAspectRatio,
-    state.showGrid,
-    state.toggleGrid,
-    shallow,
-  ]);
+  } = useHeatmapConfig();
```

Comment:
Technically, this simplification means that the toolbar is re-rendered when the domain slider and color map selector values change, but I'll make those components controlled as well. It feels wrong to retrieve half the config in the toolbar and the rest in a few child components. Ideally it should be retrieved only in child components, but that makes it impossible to create a re-usable toggler component. So I think it makes more sense to retrieve the whole config in the toolbar and re-render everything on change, and to optimise the heavy components as needed with `React.memo`.